src/Keboola/StorageApi/Metadata.php

```diff
@@ -112,6 +112,29 @@ public function postTableMetadata($tableId, $provider, $metadata)
         ));
     }
 
+    /**
+     * @param string $tableId
+     * @param string $provider
+     * @param array $metadata [['key' => 'some-key', 'value' => 'Some value'], ...]
+     * @param array $columnsMetadata ['columnName1' => [['key' => 'some-key', 'value' => 'Some value'], ...], ...]
+     * @return mixed|string
+     * @throws ClientException
+     */
+    public function postTableMetadataWithColumns($tableId, $provider, $metadata, $columnsMetadata)
+    {
+        if (!is_array($metadata) || count($metadata) === 0) {
+            throw new ClientException("Third argument must be a non-empty array of Metadata objects");
+        }
+        if (!is_array($columnsMetadata) || count($columnsMetadata) === 0) {
+            throw new ClientException("Fourth argument must be a non-empty array of Metadata objects with columns names as keys");
+        }
+        return $this->client->apiPostJson("tables/{$tableId}/metadata", array(
+            "provider" => $provider,
+            "metadata" => $metadata,
+            "columnsMetadata" => $columnsMetadata,
+        ));
+    }
```

Comment:
toto sa mě nelíbý upravil bych tu původní metodu když tam přidáš další argument `$columnsMetadata` tak to není BC a ty `columnsMetadata` sů přeca optional. Nemusí to mět pak ani vlastní test, ale enom doplnit současné. Jediná nevýhoda je, že vypadne test toho původního endpointu, ten by šel, ale udělat přes guzzle přímo.

Pokud by sas nad tím chtěl už vyhrát a dělat novů metodu tak bych udělal Options objekt tak jak to teď děláme jinde, takto přes pole ne. Ale aj tak bych zrušil ten původní form data call a posílal json.

Reply:
No, ale vraci to jinou response, proto jsem to udelal jako jinou metodu. Pac kdo to bude chtit pouzit, tak si musi predelat zpracovani odpovedi...
Takze vlastne pridanim 3. volitelneho parametru by vznikla BC.

Form-data response:
```json
[
  {
    "id": "36", "..."
  }
]

```
Json response:
```json
{
  "metadata": [
    {
      "id": "12", "..."
    }
  ],
  "columnsMetadata": {
    "id": [
      {
        "id": "14", "..."
      }
    ]
  }
}
```

Reply:
Jeste jinak... Pokud bych jen upravil parametry z 2 na 3, tak by se pouhym pouzitim 3. parametru zmenil format odpovedi a nebyla by zpetne kompatibilni, coz je dost neocekavany chovani.

Puvodni metodu `postTableMetadata` teda proto prejmenovat nemuzeme, takze zbyva maximalne tu novou `postTableMetadataWithColumns` pojmenovat nejak lip, treba `postTableMetadataBulk`.

Reply:
Jo na response sem nemyslel :facepalm: to máš pravdu. 
No tak bych udělal stejně ten options objekt. Klidně by sa to mohlo volat přes __call nebo func_get_arg a byla by to stejná metoda :thinking_face

Reply:
Když zavoláš nový endpoint místo starého tak nevrátíš tu response celú

Reply:
Tady jak bych si to asi představoval úplně bez BC:
https://github.com/keboola/storage-api-php-client/commit/2b1cd65c17c8982babde69cd04f5a20a659ed60e
Akorát ten @method typehing je dost na hovno.

Reply:
Tak sem to eště protáhl v péhapkářoch a asi je to dost unsafe pro statický analýzu :D

Reply:
Tak to nech jak to máš enom daj tu předchozí `@deprecared` s pěkným komentářem.

Reply:
🥳  upraveno